### PR TITLE
Feature/create folder modal

### PR DESCRIPTION
**Pull request by dneukirchen**
### Summary of Changes

With this PR it is possible to create new folders in the currently selected directory.

bildschirmfoto 2017-02-22 um 22 47 
25

* Created a reusable modal component with the look and feel of b2 modals
* Created create-folder-modal component (which uses the modal component)
* Created the "create-folder-form" inside the modal
* Updated the client api: implemented createFolder method
* Updated state actions and mutations with the new possibilities
* Quick Fix for the missing single resource response in the backend api and filehandler

Follow up TODOs:
* Modal styling
* Update backend api (it was just a quick fix for the resource response see #89)
* Foldernames (currently all whitespaces are stripped, perhaps it is better to replace them with '-')
* Put the create-folder-form state in vuex
* Transition effects for the modal

### Testing Instructions

* Click on create folder
* Create folder

### CODE REVIEW

File: administrator/components/com_media/controllers/api.json.php [hunk 1]
```diff
@@ -144,7 +144,7 @@ public function files()
 						$this->adapter->createFolder($name, $path);
 					}
 
-					$data = $this->adapter->getFiles($path . '/' . $name);
+					$data = $this->adapter->getFile($path . '/' . $name);
```
Comment (dgt41) [inline administrator/components/com_media/controllers/api.json.php]: You can have multiple files uploaded...

Comment (dneukirchen) [inline administrator/components/com_media/controllers/api.json.php]: But not in the current implementation if the api. The adapter calls create only one file/folder per request.

```
$this->adapter->createFile($name, $path, $mediaContent);

$this->adapter->createFolder($name, $path);
```

File: administrator/components/com_media/resources/components/toolbar/toolbar.vue [hunk 2]
```diff
@@ -2,7 +2,7 @@
     <div class="media-toolbar">
         <div class="create-wrapper">
             <div class="btn-group">
-                <button class="btn btn-success">Upload</button>
+                <button class="btn btn-success">New</button>
```
Comment (zero-24) [inline administrator/components/com_media/resources/components/toolbar/toolbar.vue:5]: No translation planed?

Comment (laoneo) [reply administrator/components/com_media/resources/components/toolbar/toolbar.vue:5]: https://github.com/joomla-projects/media-manager-improvement/issues/73

Comment (dgrammatiko) [reply administrator/components/com_media/resources/components/toolbar/toolbar.vue:5]: https://github.com/dgt41/bs4-custom-elements/blob/master/elements/alert/alert.js#L64-L76

**Pull request merged**